### ISSUE TITLE
isDeleted still tries to decode data

### ISSUE THREAD

**Issue by rjferguson21** [defenseunicorns/pepr#164]
With pepr `0.9` and a freshly `npx pepr init` hello-pepr.ts example plus the following `When`
```
When(a.Secret)
  .IsDeleted()
  .Then(request => {
    console.log('never gets here')
  });
```

Yields this error when a Secret is deleted:
```
[info]  c875659e-b14e-40c1-88ab-f25829c0b616 robinfo/configclient       Mutate request: /v1/Secret
TypeError: Cannot read properties of null (reading 'data')
    at convertFromBase64Map (/home/rob/du/pepr-bug/pepr-delete/node_modules/pepr/dist/lib.js:973:18)
    at processor (/home/rob/du/pepr-bug/pepr-delete/node_modules/pepr/dist/lib.js:1004:18)
    at mutate (/home/rob/du/pepr-bug/pepr-delete/node_modules/pepr/dist/lib.js:1160:30)
    at Layer.handle [as handle_request] (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/layer.js:95:5)
    at /home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/index.js:284:15
    at param (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/index.js:365:14)
    at param (/home/rob/du/pepr-bug/pepr-delete/node_modules/express/lib/router/index.js:376:14)
[2023-06-22T19:19:04.985Z] POST /mutate/ede9ce300a0d3e766b91d0d26075a7f2cbcca1b0f87f237074f4a51e82261e20?timeout=30s [500] 1 ms
```

**Comment by bdw617** [comment]
repoduction:

kubectl create secret generic foo
kubectl delete secret foo

**Comment by bdw617** [comment]
This was partially fixed in 0.10.0, The problem is With... will not look in the oldResource, still validating...

**Comment by bdw617** [comment]
validated fixed in 0.13.0
